### ISSUE TITLE
tf.keras.losses.categorical_hinge mentions [-1, 1] values while it works with one-hot-encoded tensor

### ISSUE THREAD

**Issue by FrankwaP** [tensorflow/tensorflow#38199]
Hi!

Please see:
https://www.tensorflow.org/api_docs/python/tf/keras/losses/categorical_hinge  
https://github.com/tensorflow/tensorflow/blob/v2.1.0/tensorflow/python/keras/losses.py#L866-L882

Both mention:
> y_true: The ground truth values. y_true values are expected to be -1 or 1. If binary (0 or 1) labels are provided they will be converted to -1 or 1.

While the code is --as expected-- a transcription of keras' one:
```
# ...
y_pred = ops.convert_to_tensor(y_pred)
y_true = math_ops.cast(y_true, y_pred.dtype)
pos = math_ops.reduce_sum(y_true * y_pred, axis=-1)
neg = math_ops.reduce_max((1. - y_true) * y_pred, axis=-1)
return math_ops.maximum(0., neg - pos + 1.)
```

And this code is meant to work with one-hot-encoded tensors. See the original discussion here: https://github.com/keras-team/keras/issues/2830

**Comment by ManishAradwad** [comment]
I'll start working on this, Thanks!

**Comment by ManishAradwad** [comment]
@FrankwaP The nightly version of docs mention 
`y_true : The ground truth values. y_true values are expected to be 0 or 1.`

**Comment by FrankwaP** [comment]
Maybe "y_true : The ground truth values in one-hot-encoded format." would add some clarity?

**Comment by gowthamkpr** [comment]
@FrankwaP CAn you be clear on what you are describing here. Thanks!

**Comment by FrankwaP** [comment]
Oh sorry!

The original/binary hinge loss expects a (n_samples,1) size vector for y_true, with -1 or 1 values. So it's a form of label encoding.

The categorical hinge loss has been adapted for N>2 classes, and now use a one-hot-encoded (n_samples,n_classes) size tensor for y_true.

So I think that specifying that we need a one-hot-encoded tensor is more explicit that specifying that the values should be 0 or 1, especially for people who also knows the binary hinge loss.
